third-party/re2/Makefile

```diff
@@ -21,6 +21,27 @@ CHPL_RE2_CXXFLAGS=-mmic
 endif
 endif
 
+
+# re2's makefile throws gcc specific flags that are not supported by the cray
+# compiler. This is intended to effectively replace the re2 makefile lines:
+#
+#     CXXFLAGS?=-Wall -O3 -g -pthread # can override
+#     RE2_CXXFLAGS?=-Wsign-compare -c -I. $(CCPCRE)  # required
+#     LDFLAGS?=-pthread
+#
+# with:
+#
+#     CXXFLAGS?= -O3
+#     RE2_CXXFLAGS?= -c -I. $(CCPCRE)
+#     LDFLAGS?=
+ifeq ($(CHPL_MAKE_TARGET_COMPILER),cray-prgenv-cray)
+CHPL_RE2_CXXFLAGS=-O3
```

Comment:
I think we weren't passing our own CXXFLAGS out of conservatism. I think it was easier to test this way. I don't think there's any problem passing them, provided things still work... I'd check a configuration including LLVM since that will set -fno-rtti.